mesheryctl/pkg/utils/auth.go

```diff
@@ -61,6 +62,37 @@ func NewRequest(method string, url string, body io.Reader) (*http.Request, error
 	return req, nil
 }
 
+// Function returns a new http response given a http request
+// Function will test the response and return any errors associated with it
+func NewResponse(req *http.Request) (*http.Response, error) {
+	client := &http.Client{}
+
+	// check status code from request, checks for issues with auth token
+	resp, err := client.Do(req)
+
+	if err != nil && resp == nil {
+		return resp, ErrFailRequest(err)
+	}
+
+	// If statuscode = 302, then we either have an expired or invalid token
+	// We return the response and correct error message
+	if resp.StatusCode == 302 {
+		return resp, InvalidToken()
```

Comment:
```suggestion
		return nil, InvalidToken()
```

let's only return `nil, error` in error cases, to not make confusion for the folks who'll be new to golang and the codebase